flow/embedded_views.h

```diff
@@ -183,6 +183,8 @@ class ExternalViewEmbedder {
  public:
   ExternalViewEmbedder() = default;
 
+  virtual bool HaveEmbeddedViewsMutated() = 0;
```

Comment:
nit: maybe name it `EmbeddedViewsWillMutateAtTheCurrentFrame`? Since this method has to be called before the mutation actually happens at the same frame, we probably want to to have the name indicate that. This method is not going to give the desired result if called after `CompositeEmbeddedView`.
I'm not good with names tho @amirh might have some thoughts.
Also, maybe document it somewhere to explain when the method should be invoked to get the desired result.

Reply:
Or maybe just `EmbeddedViewsWillMutate` and document explain the current frame part.

Reply:
I am not very satisfied with the name. having `CurrentFrame` is misleading as we could process multiple layer trees in the current frame. I added a comment saying that this needs to be called after pre-roll.

Reply:
done.